Doc/tools/static/switchers.js

```diff
@@ -0,0 +1,123 @@
+(function() {
+  'use strict';
+
+  var all_versions = {
+    '3.7': 'dev (3.7)',
+    '3.6': '3.6',
+    '3.5': '3.5',
+    '3.4': '3.4',
+    '3.3': '3.3',
+    '2.7': '2.7',
+  };
+
+  var all_languages = {
+      'en': 'English',
+      'fr': 'Français',
+  };
+
+  function build_version_select(current_version, current_release) {
+    var buf = ['<select>'];
+
+    $.each(all_versions, function(version, title) {
+      buf.push('<option value="' + version + '"');
+      if (version == current_version)
+        buf.push(' selected="selected">' + current_release + '</option>');
+      else
+        buf.push('>' + title + '</option>');
+    });
+
+    buf.push('</select>');
+    return buf.join('');
+  }
+
+  function build_language_select(current_language) {
+    var buf = ['<select>'];
+
+    $.each(all_languages, function(language, title) {
+      if (language == current_language)
+        buf.push('<option value="' + language + '" selected="selected">' +
+                 all_languages[current_language] + '</option>');
+      else
+        buf.push('<option value="' + language + '">' + title + '</option>');
+    });
+    buf.push('</select>');
+    return buf.join('');
+  }
+
+  function naviagate_if_exists(url, default_url) {
+    // check beforehand if url exists, else redirect to default_url.
+    $.ajax({
+      url: url,
+      success: function() {
+         window.location.href = url;
+      },
+      error: function() {
+         window.location.href = default_url;
+      }
+    });
+  }
+
+  function on_version_switch() {
+    var selected_version = $(this).children('option:selected').attr('value') + '/';
+    var url = window.location.href;
+    var current_language = find_language_in_url(url);
+    var current_version = find_version_in_url(url);
+    var new_url = url.replace('.org/' + current_language + current_version,
+                              '.org/' + current_language + selected_version);
+    if (new_url != url) {
+      naviagate_if_exists(new_url, 'https://docs.python.org/' +
+                          current_language + selected_version);
+    }
+  }
+
+  function on_language_switch() {
+    var selected_language = $(this).children('option:selected').attr('value') + '/';
+    var url = window.location.href;
```

Comment:
Yes I could factorize it, I actually though about it a few seconds, but I don't see how it can make code any better, in the "more readable" sense. Yes it can probably be made shorter (not even sure). But I like the readability of:
```
var new_url = url.replace('.org/' + current_language + current_version,
                          '.org/' + selected_language + current_version);
```
and
```
var new_url = url.replace('.org/' + current_language + current_version,
                          '.org/' + current_language + selected_version);
```

and factorizing it will put a new level of "indirection", making it ultimately harder to read.